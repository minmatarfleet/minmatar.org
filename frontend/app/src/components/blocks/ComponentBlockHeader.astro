---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

interface Props {
    id?:        string;
}

const {
    id,
} = Astro.props

import { marked } from 'marked';

import Flexblock from '@components/compositions/Flexblock.astro';

import CollapsableButton from '@components/blocks/CollapsableButton.astro';
import ComponentBlock from '@components/blocks/ComponentBlock.astro';
import DebugTag from '@components/blocks/DebugTag.astro';
---

<Flexblock
    gap='0'
    x-data={`{
        expanded: true,
        location_assets_count: 0,
        collapse() {
            this.expanded = false
        },
        toggle_collapse() {
            this.expanded = !this.expanded
            global_collapse = false
        },
    }`}
    x-effect="global_collapse && collapse()"
>
    <div
        class="[ sticky-header ]"
        x-bind:style="`top: ${post_header_height - 2}px;`"
    >
        <ComponentBlock
            width='narrow'
            class="[ border-bottom ]"
        >
            <CollapsableButton class="[ !block !w-full ]" x-on:click="toggle_collapse()">
                <slot name="header">
                    <DebugTag set:html={marked.parse(t('block_missing_header_slot'))} />
                </slot>
            </CollapsableButton>
        </ComponentBlock>
    </div>

    <ComponentBlock
        width='narrow'
        x-show="expanded"
        x-collapse.duration.0ms
    >
        <slot />
    </ComponentBlock>
</Flexblock>

<style lang="scss">
    .content {
        position: relative;
    }

    .sticky-header {
        position: sticky;
        top: 137px;
        z-index: var(--sticky-z-index);

        button {
            display: block;
            width: 100%;
        }
    }
</style>