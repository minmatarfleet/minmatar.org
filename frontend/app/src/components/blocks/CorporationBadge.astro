---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import type { CorporationBadgeProps } from '@dtypes/layout_components'
import { get_corporation_by_id } from '@helpers/api.eveonline/corporations'
import type { CorporationEvE } from '@dtypes/api.eveonline'
import { prod_error_messages } from '@helpers/env'

interface Props {
    corporation:    CorporationBadgeProps;
    border?:        boolean;
}

const {
    corporation,
    border = false,
} = Astro.props

let corporation_eve:CorporationEvE | null = null
let get_eve_corporation_error
if (!corporation?.name) {
    try {
        corporation_eve = await get_corporation_by_id(corporation.id)
    } catch (error) {
        get_eve_corporation_error = (prod_error_messages() ? t('get_eve_corporation_error') : error.message)
    }
}

const corporation_name = corporation?.name ?? (corporation_eve?.name ?? get_eve_corporation_error)
const size = corporation?.size ?? 'sm'
const logo_size = (size == 'sm' ? 32 : 64)
const description:string | false = corporation?.description ?? false

import { get_corporation_logo } from '@helpers/eve_image_server';
import FixedFluid from '@components/compositions/FixedFluid.astro'
import Flexblock from '@components/compositions/Flexblock.astro';

const gap = size === 'sm' ? 'var(--space-2xs)' : 'var(--space-s)'
---
<FixedFluid
    width={`${logo_size}px`}
    gap={gap}
    class="[ items-center ]"
>
    <picture class:list={{ border: border }}>
        <img loading="lazy" src={get_corporation_logo(corporation.id, logo_size)} width={logo_size} height={logo_size} />
    </picture>
    <Flexblock gap='0' class="[ w-full ]">
        {size == 'lg' ?
            <span>
                <span class="[ text-[var(--highlight)] max-w-fit ]">{corporation_name}</span>
            </span>
            <slot>
                <small>{description}</small>
            </slot>
            :
            <small class="[ max-w-fit ]">
                <a class="[ text-[var(--highlight)] ]">{corporation_name}</a>
            </small>
        }
    </Flexblock>
</FixedFluid>