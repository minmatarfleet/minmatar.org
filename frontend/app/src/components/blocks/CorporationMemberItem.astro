---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import type { CharacterKind } from '@dtypes/layout_components'

interface Props {
    member:    CharacterKind,
    corporation_id: number;
    [propName: string]: any;
}

const {
    member,
    corporation_id,
    ...attributes
} = Astro.props

import FlexInline from '@components/compositions/FlexInline.astro';

import PilotBadge from '@components/blocks/PilotBadge.astro';
import Tag from '@components/blocks/Tag.astro';
---

<FlexInline
    class="[ corporation-member-item ]"
    justification='space-between'
    {...attributes}
>
    <PilotBadge
        class="basis-[300px]"
        character_id={member.character_id}
        character_name={member.character_name}
    >
        {member.is_main &&
            <small class="truncate">{t('main_character')}</small>
        }
        {!member.is_main &&
            <small class="truncate">{member.main_character.character_name} {t('alt_character').toLowerCase()}</small>
        }
    </PilotBadge>
    {!member.exempt && !member.registered &&
        <Tag text={t('unregistered')} />
    }
    {member.exempt &&
        <Tag color='alliance-blue' text={t('exempt')} />
    }
</FlexInline>