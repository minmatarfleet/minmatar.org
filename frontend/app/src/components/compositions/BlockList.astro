---
interface Props {
    gap?:               string;
    [propName: string]: any;
}

const {
    gap = 'var(--space-s)',
    ...attributes
} = Astro.props;

delete attributes.class
---

<div class:list={['block-list', Astro.props.class]} {...attributes}>
    <slot />
</div>

<style lang="scss" define:vars={{
    block_list_gap: gap,
}}>
    .block-list {
        gap: var(--block_list_gap);
    }
</style>

<style lang="scss" is:global>
    .block-list {
        --block-list--gap: var(--space-s);

        display: flex;
        flex-direction: column;
        flex-wrap: nowrap;
        justify-content: flex-start;

        &[data-mod="centered"] {
            align-items: center;
        }

        > * {
            width: 100%;
        }
    }
</style>