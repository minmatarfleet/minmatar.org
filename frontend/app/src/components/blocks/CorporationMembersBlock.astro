---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import type { CorporationMembers } from '@dtypes/layout_components'

interface Props {
    corporation:  CorporationMembers;
    [propName: string]: any;
}

const {
    corporation,
    ...attributes
} = Astro.props

import BlockList from '@components/compositions/BlockList.astro';

import CorporationBadge from '@components/blocks/CorporationBadge.astro';
import CorporationMemberItem from '@components/blocks/CorporationMemberItem.astro';
import ComponentBlockHeader from '@components/blocks/ComponentBlockHeader.astro';
---

<div
    {...attributes}
    x-data={`{
        show_corporation() {
            return ("${corporation.corporation_id}" in corporations_filtered) && corporations_filtered[${corporation.corporation_id}].length > 0
        },
        get_members_count() {
            if ("${corporation.corporation_id}" in corporations_filtered_members_count)
                return corporations_filtered_members_count[${corporation.corporation_id}]

            return 0
        }
    }`}
    x-show="show_corporation()"
>
    <ComponentBlockHeader
        id={corporation.corporation_id.toString()}
    >
        <CorporationBadge
            slot="header"
            corporation={{
                id: corporation.corporation_id,
                name: corporation.corporation_name,
                size: 'lg',
            }}
        >
            <small
                x-text={'`${get_members_count()} ${get_members_count() != 1 ? `'+t('members').toLowerCase()+'` : `'+t('member').toLowerCase()+'`}`'}
            >
                {corporation.members.length} {corporation.members.length != 1 ? t('members').toLowerCase() : t('member').toLowerCase()}
            </small>
        </CorporationBadge>

        <BlockList
            gap='var(--space-3xs)'
            class="[ min-h-[64px] justify-center ]"
            x-show="expanded"
            x-collapse.duration.100ms
        >
            {corporation.members.map( (member) => 
                <CorporationMemberItem
                    member={member}
                    corporation_id={corporation.corporation_id}
                    x-data={`{
                        show_item() {
                            return ("${corporation.corporation_id}" in corporations_filtered) && corporations_filtered[${corporation.corporation_id}].includes(${member.character_id})
                        }
                    }`}
                    x-show="show_item()"
                />
            )}
        </BlockList>
    </ComponentBlockHeader>
</div>