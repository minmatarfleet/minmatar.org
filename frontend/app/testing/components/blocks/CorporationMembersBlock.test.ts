import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test, vi } from "vitest";
import CorporationMembersBlock from "@components/blocks/CorporationMembersBlock.astro";

import BlockList from "@components/compositions/BlockList.astro";

import CorporationBadge from "@components/blocks/CorporationBadge.astro";
import CorporationMemberItem from "@components/blocks/CorporationMemberItem.astro";
import ComponentBlockHeader from "@components/blocks/ComponentBlockHeader.astro";

vi.mock("@components/compositions/BlockList.astro");
vi.mock("@components/blocks/CorporationBadge.astro");
vi.mock("@components/blocks/CorporationMemberItem.astro");
vi.mock("@components/blocks/ComponentBlockHeader.astro");

test("CorporationMembersBlock defaults", async () => {
  vi.mocked(BlockList).mockReturnValue("BlockList_component");
  vi.mocked(CorporationBadge).mockReturnValue("CorporationBadge_component");
  vi.mocked(CorporationMemberItem).mockReturnValue(
    "CorporationMemberItem_component"
  );
  vi.mocked(ComponentBlockHeader).mockReturnValue(
    "ComponentBlockHeader_component"
  );

  const container = await AstroContainer.create();
  const result = await container.renderToString(CorporationMembersBlock, {
    props: {
      corporation: {
        corporation_id: 2356,
        members: [
          {
            character_id: 12,
          },
        ],
      },
    },
  });

  // expect(result).toMatchSnapshot();
});
