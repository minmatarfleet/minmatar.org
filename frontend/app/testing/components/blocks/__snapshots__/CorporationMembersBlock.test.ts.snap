// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CorporationMembersBlock defaults 1`] = `
"<div x-data="{
        show_corporation() {
            return (&#34;2356&#34; in corporations_filtered) &#38;&#38; corporations_filtered[2356].length > 0
        },
        get_members_count() {
            if (&#34;2356&#34; in corporations_filtered_members_count)
                return corporations_filtered_members_count[2356]

            return 0
        }
    }" x-show="show_corporation()"> ComponentBlockHeader_component </div>"
`;
